code/modules/projectiles/projectile.dm

```diff
@@ -117,6 +127,17 @@
 		return 0
 	if(alwayslog)
 		add_attack_logs(firer, target, "Shot with a [type]")
+	//If the projectile has knockback AND EITHER breaching OR the target is a mob, throw the target
+	//OR If the projectile has breaching AND the target is an object, throw the target
+	//Should the projectile hit a mob without having knockback or an object without being breaching, do not throw the target.
+	if((knockback && (breaching || isliving(target))) || (breaching && isobj(target)))
```

Comment:
Can you make this into a proc to make the current proc a bit more readable?
`try_apply_knockback(...)` or something among those lines will work